speech/google/cloud/speech/sample.py

```diff
@@ -51,17 +55,21 @@ class Sample(object):
     default_encoding = Encoding.FLAC
     default_sample_rate = 16000
 
-    def __init__(self, content=None, source_uri=None,
+    def __init__(self, content=None, source_uri=None, stream=None,
                  encoding=None, sample_rate=None, client=None):
         self._client = client
 
-        no_source = content is None and source_uri is None
-        both_source = content is not None and source_uri is not None
+        no_source = content is None and source_uri is None and stream is None
+        both_source = (content is not None
+                       and source_uri is not None
+                       and stream is not None)
```

Comment:
This isn't what you want. You are checking `000` and `111` among the 8 possible null/non-null triples. You are accidentally allowing `011`, `101` and `110`. Instead:

```python
non_null = (int(content is not None) +
            int(source_uri is not None) +
            int(stream is not None))
if non_null != 1:
    raise ValueError('...')
```

Also "both" would no longer apply here.

(ISTM your unit tests that test those situations should fail, so maybe I am misunderstanding something.)